src/pandroid/app/src/main/java/com/panda3ds/pandroid/app/GameActivity.java

```diff
@@ -0,0 +1,39 @@
+package com.panda3ds.pandroid.app;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Handler;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+
+import com.panda3ds.pandroid.AlberDriver;
+import com.panda3ds.pandroid.C;
+import com.panda3ds.pandroid.app.BaseActivity;
+import com.panda3ds.pandroid.view.PandaGlSurfaceView;
+
+public class GameActivity extends BaseActivity {
+    private PandaGlSurfaceView pandaSurface;
+
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        pandaSurface = new PandaGlSurfaceView(this);;
+        setContentView(pandaSurface);
+
+        Intent intent = getIntent();
+        if(!intent.hasExtra(C.EXTRA_PATH)){
+            Toast.makeText(this, "INVALID ROM PATH", Toast.LENGTH_LONG).show();
```

Comment:
If your C.java class is for string literals, perhaps this error string should be moved there (and also not capitalized)